internal/jimm/role/role_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 Canonical.
+package role
+
+import (
+	"context"
+	"sort"
+	"strconv"
+	"testing"
+	"time"
+
+	cofga "github.com/canonical/ofga"
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+	jimmtest "github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type roleManagerSuite struct {
+	manager    *roleManager
```

Comment:
also we don't necessarily need this field on the suite. Calling the NewRoleManager constructor takes one line in each test and we're not using any of the role package internals really so i don't think this is a compelling reason for whitebox testing here..

Reply:
This is a simple example, but some services may benefit from testing private functions for edge behaviour or perhaps there are variables that can be tweaked. Things like this.

Reply:
I think better to do whitebox testing on a case by case basis where needed. Here, as in most of the rest of the code, my suggestion would be to keep it blackbox and using a role_testing package. I liked the idea of being deliberate with how you structure these black vs white tests because it prevents you from accidentally using something that is not exposed and feels like better practice.